sdk/search/Azure.Search.Documents/src/Generated/Models/SynonymMap.cs

```diff
@@ -11,8 +11,13 @@ namespace Azure.Search.Documents.Models
     public partial class SynonymMap
     {
         /// <summary> Initializes a new instance of SynonymMap. </summary>
-        public SynonymMap()
+        /// <param name="name"> The name of the synonym map. </param>
+        /// <param name="synonyms"> A series of synonym rules in the specified synonym map format. The rules must be separated by newlines. </param>
+        public SynonymMap(string name, string synonyms)
         {
+            Name = name;
+            Format = "solr";
```

Comment:
While this has a default value, it should still be a parameter with a default because it could be changed.

Reply:
I think it's defined as constant not a default value.

Reply:
It's an enum: https://github.com/Azure/azure-rest-api-specs/blob/8919f8072121d11f0bb61f1c5d02f2f4063d4f48/specification/search/data-plane/Microsoft.Azure.Search.Service/stable/2019-05-06/searchservice.json#L5181-L5191

Though, it's odd it doesn't generate an enum.

Reply:
Enum with a single value has a special meaning. It basically means it's a constant.

Swagger is great :)

Reply:
They have indicated they could add values in the future, but I suppose we could overload it then. But if we expose the model as public (which we do), how would you know to define the old behavior? Seems that special casing for a single-value enum may not always work.